tests/lean/interactive/hover.lean

```diff
@@ -215,3 +215,24 @@ example : α → α := by my_intro x; assumption
                              --^ textDocument/hover
 example : α → α := by my_intro _; assumption
                              --^ textDocument/hover
+
+example : Nat → True := by
+  intro x
+      --^ textDocument/hover
+  cases x with
+  | zero => trivial
+  --^ textDocument/hover
+  --v textDocument/hover
+  | succ x => trivial
+       --^ textDocument/hover   -- TODO: broken
+
+example : Nat → True := by
+  intro x
+      --^ textDocument/hover
+  induction x with
+  | zero => trivial
+  --^ textDocument/hover
+       -- TODO: the next two are broken
+       --v textDocument/hover
+  | succ _ ih => exact ih
+         --^ textDocument/hover
```

Comment:
@Kha Setting aside the `_` binder, do you have any ideas why the hover on `ih` doesn't work? Go-to-def will *target* the correct span of the `ih` binder (for example if you click on the `ih` in `exact ih`), but directly hovering the binder does not reveal the type and instead does the fallback thing of tellling you about the induction alt LHS syntax `| @foo x y z`. But it seems very similar to `intro x`, which does work.

Reply:
Ahh, it's the old "how tree-structured should the info tree actually be" discussion - note that in the info tree, the pattern variable infos are outside of the tactic info for that case. Perhaps we should just resurrect https://github.com/Kha/lean4/commit/0bd864deca82ee865a2e5134d5c1f6105f5a701c#diff-83bd7ca17a546fdf495692949c7412fc3a059f292c59e667898804da030dba22L157-L158 again, it's not exactly a terrible complication.

Reply:
What exactly is the nesting relation we need here? The tactic info for `induction` should be huge here, covering the `induction` all the way until the end of the proof, so I'm not sure how it could have migrated out.

Reply:
The current server code assumes that info tree nodes' ranges are contained within their parent range and that sibling ranges are disjoint (apart from duplicates). But as I said, there is a tactic info for each case that spans over all pattern variables yet does not contain their infos.

Reply:
oh I see, is this the `| foo x y z =>` tactic info that we use to show the initial and final states for the case?

Reply:
Yes, exactly

Reply:
I fixed the info tree handling, so now these hovers work as expected. (There was also a similar bug where the `x` in `induction x with ...` was being blocked by the inner tactic info we use to show the original goal state outside the alternatives, which has also been fixed.)